File: src/utils/crypto.js
import CryptoJS from 'crypto-js'

// 使用固定的16字节密钥
const SECRET_KEY = 'ABCDEF0123456789'

export const encrypt = (data) => {
  try {
    // 将数据转换为JSON字符串
    const jsonStr = JSON.stringify(data)
    
    // 使用AES-ECB模式加密
    const encrypted = CryptoJS.AES.encrypt(jsonStr, SECRET_KEY, {
      mode: CryptoJS.mode.ECB,
      padding: CryptoJS.pad.Pkcs7
    })
    
    // 返回Base64编码的密文
    return encrypted.toString()
  } catch (error) {
    console.error('加密失败:', error)
    return null
  }
}

export const decrypt = (encryptedData) => {
  try {
    // AES 解密
    const bytes = CryptoJS.AES.decrypt(encryptedData, SECRET_KEY)
    const decrypted = bytes.toString(CryptoJS.enc.Utf8)
    return JSON.parse(decrypted)
  } catch (error) {
    console.error('解密失败:', error)
    return null
  }
} 

File: backend/utils/crypto.py
from Crypto.Cipher import AES
from Crypto.Util.Padding import unpad
import base64
import json

# 使用与前端相同的16字节密钥
SECRET_KEY = b'ABCDEF0123456789'

def decrypt_data(encrypted_data):
    try:
        # Base64解码
        encrypted_bytes = base64.b64decode(encrypted_data)
        
        # 创建AES-ECB模式的cipher
        cipher = AES.new(SECRET_KEY, AES.MODE_ECB)
        
        # 解密并去除填充
        decrypted_bytes = unpad(cipher.decrypt(encrypted_bytes), AES.block_size)
        
        # 转换为字符串并解析JSON
        decrypted_str = decrypted_bytes.decode('utf-8')
        return json.loads(decrypted_str)
    except Exception as e:
        print(f"解密失败: {e}")
        print(f"加密数据: {encrypted_data}")
        return None 